controllers/workspace/provision/deployment.go

```diff
@@ -337,6 +365,60 @@ func getPods(workspace *dw.DevWorkspace, client runtimeClient.Client) (*corev1.P
 	return pods, nil
 }
 
+func getDevWorkspaceConfigmaps(namespace string, client runtimeClient.Client) (*v1alpha1.PodAdditions, []corev1.EnvVar, error) {
+	configmaps := &corev1.ConfigMapList{}
+	if err := client.List(context.TODO(), configmaps, k8sclient.InNamespace(namespace), k8sclient.MatchingLabels{
+		constants.DevWorkspaceMountLabel: "true",
+	}); err != nil {
+		return nil, nil, err
+	}
+	podAdditions := &v1alpha1.PodAdditions{}
+	additionalEnvVars := []corev1.EnvVar{}
+	for _, configmap := range configmaps.Items {
+		mountAs := configmap.Annotations[constants.DevWorkspaceMountAsAnnotation]
+		if mountAs == "env" {
```

Comment:
I would assume this would be done via [`envFrom`](https://kubernetes.io/docs/tasks/configure-pod-container/configure-pod-configmap/#configure-all-key-value-pairs-in-a-configmap-as-container-environment-variables). If someone is creating a configmap and specifically marking it as "mount this as environment variables", this is the k8s analaogue I'd choose.

Reply:
Do you mean linking to the configmap through K8s mechanism(instead of propagating it manually),
Not changing the expected value for this annotation?